### ISSUE TITLE
Visibility of values is poor at lower resolutions/bitrates

### ISSUE THREAD

**Issue by schwjm** [akdb/pso-tracker#3]
Due to the placement of text, the current fonts, and the current font shadow (overlaid blur) strategy, certain numbers are not super clear. In general, the readability probably could be improved but some specific cases stand out:

- Zonde 1
- MST +8
![image](https://user-images.githubusercontent.com/20408541/58374751-1f6a3c80-7f09-11e9-8438-e50e685113d9.png)

Ideally, can this be significantly improved without extreme solutions like shrinking the icons significantly across the board or moving the text entirely outside the cell. Better fonts, better contrast strategies, what can be done?

Notable test parameters: 300x256 window size, 1800 kbps stream bitrate

**Comment by eleriaqueen** [comment]
Not sure how you handle those numbers in project (Browser renders the font? I don't see a SVG "numbers sprite sheet" or something of the sort), but you could try adding high contrast outlines to each number (ideally black), maybe increase digits size.
Are there dropdown shadows under these numbers ? You might want to turn them off as it might create artifacts in the video and actually make them less readable (especially if the shadows are smooth).

**Comment by schwjm** [comment]
It's inserting `text` SVG elements. See https://github.com/akdb/pso-tracker/blob/25284d701034a0e3cc97a856f18d3653d35d22f8/src/modules/Palette.js#L313-L323

[`plain()`](https://svgjs.com/docs/3.0/elements/#svg-text) creates a primitive text element without styling.

SVG apparently doesn't offer shadows directly like HTML+CSS does so the common workaround to make a "shadow" or "glow" is to make a second text element of the shadow color and blur it. I believe it's important to have some sort of contrast between the white digits and the icons around them, white on yellow without the black would be even harder to read. The point about artifacting I get though, it might be better to blur less or not at all and just use a thick stroke on the `textGlow` element.

In general I'm not satisfied with the font I'm using but it was the least worst of what I tried, my requirements were only that realistic values would fit within the hexagon while still being readable (tougher than it might seem because some fonts like to space characters out a lot and support for adjusting that in SVG is apparently lacking as I tried it).

**Comment by eleriaqueen** [comment]
Yes, you're right, there needs to be a pretty high/good contrast between digits and what's under, I think no blur at all would work, varying the stroke size on that textGlow element could be the key.

Hmm, I don't have knowledge of really good fonts, but I usually use different flavours of Google's "Roboto" font in my projects, it's not stylised though.
You can see specimens and also download it (not straight-forward) over at : https://fonts.google.com/specimen/Roboto

**Comment by schwjm** [comment]
I can try Roboto again. I tried it before and don't remember if it had a spacing issue or if it was just the non-standardness that drove me back toward Arial. (Of course I referenced Dotum in the end which I think might be only available on Windows by default).

I was shying away from custom fonts because of trying to keep this as compact and offline as possible (I probably should remember to document that goal). I don't mind compromising the download size if I can get a font that works well. One instance of Roboto itself would still only be the third largest file and the total for the release bundle would be under 1MB still.

**Comment by schwjm** [comment]
I tried the following with a thicker glow stroke. I kept the blur in so that it would spread out as before, just adding more of a black border seems to help the blur be effective without just being a solid wall. The font family doesn't seem to make a huge difference, and Arial remains in my opinion the easiest to read at the low resolution/bitrate, and while Roboto looks nicer it doesn't look THAT different for single digits and is a bit skinnier. I also tried Consolas which is okay.

Here's some captures of recording 1800 kbps video with similar sizes to the original screenshot.
Arial:
![arial](https://user-images.githubusercontent.com/20408541/58833975-d3f61380-8617-11e9-867d-9f71ebc4ffce.PNG)
Roboto:
![roboto](https://user-images.githubusercontent.com/20408541/58833977-d5274080-8617-11e9-9890-9dbc01598145.PNG)
Consolas:
![consolas](https://user-images.githubusercontent.com/20408541/58833981-d6f10400-8617-11e9-8908-4cfc88e385df.png)

My inclination is to move the text over to the corner (as it is done with increment buttons), and even move it over the cell border a little bit.

**Comment by eleriaqueen** [comment]
I think the thicker glow looks way worse. 
Have you tried using a black filled rounded rectangle with maybe 50 to 90 % transparency as a way to highlight text elements ? I mean something that would sit behind text elements but over the BG.

**Comment by schwjm** [comment]
I haven't tried a rounded rectangle but I'm uncertain of how I would get that to dynamically size, or maybe static is enough (for most intents and purposes, you're never going to see a double digit tech in a normal run, but what about HP which can be wide or very wide).

I did try opacity on the glow text and it looked goofy due to how alpha blending worked, if I recall correctly. I had to keep it at 25% transparency to have the contrast stay good enough as it scaled down so it seemed pointless.
